File: 590. N-ary Tree Postorder Traversal/solution.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val, children):
        self.val = val
        self.children = children
"""


class Solution:
    def postorder(self, root: 'Node') -> List[int]:
        res = []

        def helper(node):
            if not node:
                return
            for child in node.children:
                helper(child)
            res.append(node.val)
        helper(root)
        return res

File: 590. N-ary Tree Postorder Traversal/solution.js
/**
 * // Definition for a Node.
 * function Node(val,children) {
 *    this.val = val;
 *    this.children = children;
 * };
 */
/**
 * @param {Node} root
 * @return {number[]}
 */
var postorder = function(root) {
  const res = [];
  function order(root) {
    if (root) {
      const cd = root.children;
      cd.forEach(ele => order(ele));
      res.push(root.val);
    }
  }
  order(root);
  return res;
};
